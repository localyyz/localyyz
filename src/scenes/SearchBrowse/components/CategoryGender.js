import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";

// third party
import { inject, observer } from "mobx-react/native";

// custom
import { Colours, Sizes, Styles } from "localyyz/constants";
import { capitalize } from "localyyz/helpers";

// constants
const GENDERS = [
  { id: "female", label: "Woman", filterId: "woman" },
  { id: "male", label: "Man", filterId: "man" },
  { id: "unisex", label: "Unisex", filterId: "unisex" }
];
const GENDER_MAPPING = Object.assign(
  {},
  ...GENDERS.map(gender => ({ [gender.id]: gender }))
);

@inject(stores => ({
  gender: stores.searchStore.gender,
  setGender: stores.searchStore.setGender,
  userGender: stores.userStore.gender
}))
@observer
export default class CategoryGender extends React.Component {
  constructor(props) {
    super(props);

    // auto select gender if not given from user settings
    !props.gender
      && GENDER_MAPPING[props.userGender]
      && props.setGender(GENDER_MAPPING[props.userGender]);
  }

  render() {
    return (
      <View style={styles.container}>
        {GENDERS.map((gender, i) => (
          <View style={styles.button} key={`gender-${gender.id}-${i}`}>
            <TouchableOpacity onPress={() => this.props.setGender(gender)}>
              <View
                style={[
                  styles.option,
                  this.props.gender
                    && this.props.gender.id == gender.id
                    && styles.active
                ]}>
                <Text style={styles.label}>{capitalize(gender.label)}</Text>
              </View>
            </TouchableOpacity>
          </View>
        ))}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...Styles.Horizontal,
    ...Styles.EqualColumns,
    padding: Sizes.InnerFrame
  },

  button: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  },

  option: {
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: Sizes.InnerFrame / 4,
    paddingHorizontal: Sizes.OuterFrame * 2 / 3,
    borderRadius: Sizes.InnerFrame
  },

  active: {
    backgroundColor: Colours.Action
  },

  label: {
    ...Styles.Text,
    ...Styles.Emphasized
  }
});
