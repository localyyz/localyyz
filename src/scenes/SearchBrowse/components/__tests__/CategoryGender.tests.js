import React from "react";
import Renderer from "react-test-renderer";

import CategoryGender from "../CategoryGender";

describe("CategoryGender", () => {
  it("should render properly", () => {
    let render = Renderer.create(<CategoryGender.wrappedComponent />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  it("should render properly with user gender", () => {
    const PROPS = {
      userGender: "female",
      setGender: jest.fn()
    };

    let render = Renderer.create(
      <CategoryGender.wrappedComponent {...PROPS} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it("should render properly with selected gender", () => {
    const PROPS = {
      gender: "female",
      setGender: jest.fn()
    };

    let render = Renderer.create(
      <CategoryGender.wrappedComponent {...PROPS} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
