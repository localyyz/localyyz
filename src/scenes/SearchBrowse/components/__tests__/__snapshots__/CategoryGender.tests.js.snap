// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryGender should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Women
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Men
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`CategoryGender should render properly with selected gender 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Women
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Men
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`CategoryGender should render properly with user gender 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Women
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "paddingHorizontal": 16.666666666666668,
              "paddingVertical": 3.75,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Men
        </Text>
      </View>
    </View>
  </View>
</View>
`;
